main/src/ca/uwaterloo/flix/language/phase/jvm/GenAnonymousClasses.scala

```diff
@@ -118,9 +118,10 @@ object GenAnonymousClasses {
     */
   private def compileMethod(currentClass: JvmType.Reference, method: JvmMethod, cloName: String, classVisitor: ClassWriter): Unit = method match {
     case JvmMethod(ident, fparams, tpe, _, _) =>
-      val methodType = MonoType.Arrow(fparams.map(_.tpe), tpe)
-      val closureAbstractClass = JvmOps.getClosureAbstractClassType(methodType)
-      val functionInterface = JvmOps.getFunctionInterfaceType(methodType)
+      val erasedArgs = fparams.map(_.tpe).map(JvmOps.getErasedJvmType)
```

Comment:
So we still do some erasure here, right?

Reply:
arguments are erased in the backend, not the eraser (this is difficult to change since the types are used in the local stack). JvmMethod return types are erased in the backend also since they type is used to generate the signature for the subclass.